modules/framework/galasa-parent/dev.galasa.framework.resource.management/src/main/java/dev/galasa/framework/resource/management/internal/MonitorConfiguration.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.framework.resource.management.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+import dev.galasa.framework.spi.FrameworkException;
+
+public class MonitorConfiguration {
+    
+    private String stream;
+    private List<Pattern> includesRegexList;
+    private List<Pattern> excludesRegexList;
+
+    public MonitorConfiguration(String stream, List<String> includesRegexList, List<String> excludesRegexList) throws FrameworkException {
+        this.stream = stream;
+        this.includesRegexList = convertListToPatternList(includesRegexList);
+        this.excludesRegexList = convertListToPatternList(excludesRegexList);
+    }
+
+    public String getStream() {
+        return stream;
+    }
+
+    public List<Pattern> getIncludesRegexList() {
+        return includesRegexList;
+    }
+
+    public List<Pattern> getExcludesRegexList() {
+        return excludesRegexList;
+    }
+
+    private List<Pattern> convertListToPatternList(List<String> regexList) throws FrameworkException {
+        List<Pattern> patternList = new ArrayList<>();
+        for (String pattern : regexList) {
+            String patternToAdd = pattern;
+
+            // Patterns like '*' and '*mysuffix' are not valid regex patterns alone, so convert them into valid
+            // patterns like '.*' and '.*mysuffix' to accept any character (.) zero or more times (*) in prefixes.
+            if (pattern.startsWith("*")) {
+                patternToAdd = "." + pattern;
+            }
```

Comment:
Is this necessary? If we say we only expect valid regex patterns, then this won't be needed since a pattern like `*myMonitorClass` isn't valid.

Reply:
Think it depends if we want to support glob matching or regex. Seems that glob matches should be sufficient so supporting *.x.y.z should probably match .*[.]x[.]y[.]z ?

Reply:
glob matches are easier for people to get right

Reply:
BUT. It's a bit odd only processing the first 'glob' pattern and leaving the others alone. And not converting '.' in the input to '[.]' in the regex.
Why only do this for the startsWith() string, rather than all the way through ?

Reply:
Thanks, switched to glob matching in https://github.com/galasa-dev/galasa/pull/236/commits/fa36d081355b9cd4e5ef366e28cde2f8089f0b53